crates/core/transaction/src/is_action.rs

```diff
@@ -329,10 +329,44 @@ impl IsAction for Swap {
         });
 
         ActionView::Swap(match plaintext {
-            Some(swap_plaintext) => SwapView::Visible {
-                swap: self.to_owned(),
-                swap_plaintext,
-            },
+            Some(swap_plaintext) => {
+                // If we can find a matching BSOD in the TxP, use it to compute the output notes
+                // for the swap.
+                let bsod = txp
+                    .batch_swap_output_data
+                    .iter()
+                    // This finds the first matching one; there should only be one
+                    // per trading pair per block and we trust the TxP provider not to lie about it.
+                    .find(|bsod| bsod.trading_pair == swap_plaintext.trading_pair);
+
+                let (output_1, output_2) = match bsod.map(|bsod| swap_plaintext.output_notes(bsod))
+                {
+                    Some((output_1, output_2)) => {
+                        (Some(txp.view_note(output_1)), Some(txp.view_note(output_2)))
+                    }
+                    None => (None, None),
+                };
```

Comment:
The output notes of the swap are determined _entirely_ by (1) the `SwapPlaintext` (committed to by the swap commitment and (2) the `BatchSwapOutputData`.  This means that to get the output notes, we should supply a BSOD in the perspective, then recompute them.

Compared to trying to insert the output notes manually, this has two advantages:

1. It correctly handles the case where the swap tx has been accepted on chain but the claim has not yet been submitted. In that case, even though we don't know the swap claim tx (it doesn't exist), we can still show the output notes.
2. It allows rendering extra information (if we want), like the % share of the batch the user's swap was.